graphql/context_operation_test.go

```diff
@@ -106,4 +106,31 @@ func TestCollectAllFields(t *testing.T) {
 		require.NotEqual(t, collected[0], collected[1])
 		require.Equal(t, collected[0].Name, collected[1].Name)
 	})
+
+	t.Run("collect fragments with same field name and different alias", func(t *testing.T) {
+		ctx := testContext(ast.SelectionSet{
+			&ast.InlineFragment{
+				TypeCondition: "ExampleTypeA",
+				SelectionSet: ast.SelectionSet{
+					&ast.Field{
+						Name:             "fieldA",
+						Alias:            "fieldA",
+						ObjectDefinition: &ast.Definition{Name: "ExampleTypeA"},
+					},
+					&ast.Field{
+						Name:             "fieldA",
+						Alias:            "fieldA Alias",
+						ObjectDefinition: &ast.Definition{Name: "ExampleTypeA"},
+					},
+				},
+				ObjectDefinition: &ast.Definition{Name: "Interface"},
```

Comment:
```suggestion
				ObjectDefinition: &ast.Definition{Name: "ExampleType", Kind: ast.Interface},
```
Can we name it something else? `Interface` name can confuse people with `interface` kind, lets name it `ExampleType` or something else.

Reply:
Renamed to ExampleType.